packages/gatsby/cache-dir/root.js

```diff
@@ -145,12 +145,31 @@ const ConditionalFastRefreshOverlay = ({ children }) => {
   return <React.Fragment>{children}</React.Fragment>
 }
 
-export default () => (
-  <ConditionalFastRefreshOverlay>
-    <StaticQueryStore>{WrappedRoot}</StaticQueryStore>
-    {process.env.GATSBY_EXPERIMENTAL_QUERY_ON_DEMAND &&
-      process.env.GATSBY_QUERY_ON_DEMAND_LOADING_INDICATOR === `true` && (
-        <LoadingIndicatorEventHandler />
-      )}
-  </ConditionalFastRefreshOverlay>
-)
+class RootComponent extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      hasRendered: false,
+    }
+  }
+  componentDidMount() {
+    this.setState({ hasRendered: true })
+  }
+  render() {
+    if (this.state.hasRendered) {
+      return (
+        <ConditionalFastRefreshOverlay>
```

Comment:
The overlays are client-only so we don't render them initially to avoid React saying there's hydration problems